docs/extending-dream2nix.md

```diff
@@ -0,0 +1,56 @@
+# Extending dream2nix with external translators, builders etc.
+
+`dream2nix` can be extended while you are `init`ializing it.
+This can be done in a few ways. For extending, you need to
+utilize the `config.extra` option of the dream2nix config.
+
+## Declare `extra`s from a nix file
+
+```nix
+dream2nix.lib.init {
+  config.extra = ./nix/d2n/extras.nix;
+}
+```
+this file should look like this:
+```nix
+{dlib, lib, config, ...}:
+{
+  subsystems.rust.translators.new = ./rust-translator.nix;
+  # you can declare modules using functions here
+  fetchers.ipfs = {...}: {/* fetcher attrs */};
+}
+```
+
+## Declare `extra`s as an attribute set
+
+```nix
+dream2nix.lib.init {
+  config.extra = {
+    subsystems = {
+      # add new modules
+      ruby.discoverers.default = ./nix/d2n/ruby/discoverer.nix;
+      ruby.translators.bundix = ./nix/d2n/ruby/bundix.nix;
+      # existing modules can be overrided
+      rust.builders.crane = ./nix/d2n/rust/crane.nix;
```

Comment:
Not sure where I can put that comment here, but probably we should explain how you should delegate to other builders now – I noticed now that `buildPackageWithOtherBuilder` wasn't actually removed in your PR but is already gone on master, but maybe some remark in the template on how to refer to another builder would be useful?

Reply:
We shouldn't need to explain this here, it should be explained for docs on implementing a translator.